src/components/icon/icon.test.jsx

```diff
@@ -0,0 +1,117 @@
+        expect(tree.props.className).toContain(`size-${size}`);
+      });
+    });
+  });
+
+  describe('focusable API', () => {
+    it('defaults false', () => {
+      const tree = renderer.create((
+        <Icon name={'foobar'} />
+      )).toJSON();
+      expect(tree.props.focusable).toEqual(false);
+    });
+
+    it('sets to true', () => {
+      const tree = renderer.create((
+        <Icon name={'foobar'} focusable={true} />
+      )).toJSON();
+      expect(tree.props.focusable).toEqual(true);
+    });
+  });
+
+  describe('className API', () => {
+    describe('fill', () => {
+      it('sets all fill colors', () => {
+        const colors = ['primary', 'action', 'highlight', 'caution'];
+        colors.forEach((color) => {
+          const tree = renderer.create((
+            <Icon name={'foobar'} fill={color} />
+          )).toJSON();
+          expect(tree.props.className).toContain('icon-base');
+          expect(tree.props.className).toContain(`fill-${color}`);
+          expect(tree.props.className).not.toContain('stroke');
+          expect(tree.props.className).not.toContain('color');
+        });
+      });
+    });
+
+    describe('stroke', () => {
+      it('sets stroke colors turning off fills and color', () => {
+        const colors = ['primary', 'action', 'highlight', 'caution'];
+        colors.forEach((color) => {
+          const tree = renderer.create((
+            <Icon name={'foobar'} stroke={color} />
+          )).toJSON();
+          expect(tree.props.className).toContain('icon-base');
+          expect(tree.props.className).toContain(`stroke-${color}`);
+          expect(tree.props.className).toContain('fill-none');
+          expect(tree.props.className).toContain('color-transparent');
+        });
+      });
+    });
+
+    describe('currentColor', () => {
+      it('sets all font colors', () => {
+        const colors = ['primary', 'action', 'highlight', 'caution'];
+        colors.forEach((color) => {
+          const tree = renderer.create((
+            <Icon name={'foobar'} currentColor={color} />
+          )).toJSON();
+          expect(tree.props.className).toContain('icon-base');
+          expect(tree.props.className).toContain(`color-${color}`);
+        });
+      });
+    });
+
+    describe('combining colors', () => {
+      it('can combine fill, stroke, and currentColor', () => {
+        const colors = ['primary', 'action', 'highlight', 'caution'];
+        colors.forEach((color) => {
+          const tree = renderer.create((
+            <Icon name={'foobar'} fill={color} stroke={color} currentColor={color} />
+          )).toJSON();
+          expect(tree.props.className).toContain('icon-base');
+          expect(tree.props.className).toContain(`fill-${color}`);
+          expect(tree.props.className).toContain(`color-${color}`);
+          expect(tree.props.className).toContain(`stroke-${color}`);
+        });
+      });
+    });
+  });
+
+  describe('title API', () => {
```

Comment:
It looks like this does not work. Most of the internet is saying that an SVG does not support the title attribute.

Reply:
https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title

https://www.w3.org/TR/SVG11/struct.html#DescriptionAndTitleElements

Reply:
Neither of those links support our implementation in this PR. 

If you read closely, those are talking about a `<title>` element instead of a `title=` attribute. 

Am I misreading it? Does the attribute translate to the element under the hood?

Reply:
Oh, @roblevintennis , I think this is actionable. I do not see the title working in the deployment page.

Reply:
@juanca yeah, that was bogus to put as an attribute 🤦‍♂ 


13ee9b2 should fix it...I tried it locally and see the tooltip on hover (I just typed it in the live edit of styleguidist)

Ima add another commit with it like that way in one of the .md example too...but as of ^ commit it should at least be available for your input explorations